src/main/java/com/exasol/adapter/dynamodb/mapping/TableMappingDefinition.java

```diff
@@ -4,32 +4,45 @@
 import java.util.Collections;
 import java.util.List;
 
+import com.exasol.dynamodb.resultwalker.AbstractDynamodbResultWalker;
+
 /**
  * Definition of a table mapping from DynamoDB table to Exasol Virtual Schema. Each instance of this class represents a
  * table in the Exasol Virtual Schema. Typically it also represents a DynamoDB table. But it can also represent the data
  * from a nested list or object. See {@link #isRootTable()} for details.
  */
 public class TableMappingDefinition {
     private final String exasolName;
-    private final boolean isRootTable;
     private final List<AbstractColumnMappingDefinition> columns;
+    private final AbstractDynamodbResultWalker pathToNestedTable;
 
-    private TableMappingDefinition(final String exasolName, final boolean isRootTable,
-            final List<AbstractColumnMappingDefinition> columns) {
+    private TableMappingDefinition(final String exasolName, final List<AbstractColumnMappingDefinition> columns,
+            final AbstractDynamodbResultWalker pathToNestedTable) {
         this.exasolName = exasolName;
-        this.isRootTable = isRootTable;
         this.columns = columns;
+        this.pathToNestedTable = pathToNestedTable;
+    }
+
+    /**
+     * Gives an instance of the Builder for {@link TableMappingDefinition}. This version of the builder is used for root
+     * tables.
+     *
+     * @param destName Name of the Exasol table
+     * @return {@link TableMappingDefinition.Builder}
+     */
+    public static Builder builder(final String destName) {
```

Comment:
Maybe you should also somehow show in the names that the methods are used for different purposes.